From 6ff1f06c414631c0b74e3cd318b68adbb0abdd9c Mon Sep 17 00:00:00 2001
From: Elizabeth Figura <zfigura@codeweavers.com>
Date: Mon, 20 Oct 2025 19:29:48 -0500
Subject: [PATCH 16/48] ntdll/tests: Test NtQueryAttributesFile() with a root
 directory.

(cherry picked from commit 6f727b53bc48d386e526e1dd1e8b445ac6c2b730)

CW-Bug-Id: #26492
---
 dlls/ntdll/tests/file.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/dlls/ntdll/tests/file.c b/dlls/ntdll/tests/file.c
index 594fc4b673f..01c3f84e45e 100644
--- a/dlls/ntdll/tests/file.c
+++ b/dlls/ntdll/tests/file.c
@@ -359,6 +359,7 @@ static void open_file_test(void)
     NTSTATUS status;
     HANDLE dir, root, handle, file;
     WCHAR path[MAX_PATH], tmpfile[MAX_PATH];
+    FILE_BASIC_INFORMATION info;
     BYTE data[1024];
     OBJECT_ATTRIBUTES attr;
     IO_STATUS_BLOCK io;
@@ -398,6 +399,11 @@ static void open_file_test(void)
     ok( !status, "open %s failed %lx\n", wine_dbgstr_w(nameW.Buffer), status );
     CloseHandle( handle );
 
+    status = pNtQueryAttributesFile( &attr, &info );
+    todo_wine ok( !status, "query %s failed %lx\n", wine_dbgstr_w(nameW.Buffer), status );
+    if (!status)
+        ok( info.FileAttributes == FILE_ATTRIBUTE_DIRECTORY, "got %#lx\n", info.FileAttributes );
+
     /* try uppercase name */
     for (i = len; path[i]; i++) if (path[i] >= 'a' && path[i] <= 'z') path[i] -= 'a' - 'A';
     status = pNtOpenFile( &handle, GENERIC_READ, &attr, &io,
-- 
2.53.0

